server/src/app/user/user.service.impl.ts

```diff
@@ -34,41 +40,26 @@ export class UserServiceImpl implements UserService {
     }
 
     async increasePoint(id: number, amount: number): Promise<User> {
-        let user: User;
-        user = await this.userRepository.findById(id);
+        const user = await this.userRepository.findById(id);
         if (user === undefined) {
-            throw new Error('user with the id is not found');
+            throw new NotFoundException('user with the id is not found');
         }
 
         return user.increasePoint(amount);
     }
 
     async decreasePoint(id: number, amount: number): Promise<User> {
-        let user: User;
-        user = await this.userRepository.findById(id);
+        const user = await this.userRepository.findById(id);
         if (user === undefined) {
-            throw new Error('user with the id is not found');
-        }
-
-        let errors: Error[];
-        errors = user.canDecreasePoint(amount);
-        if (errors.length !== 0) {
-            throw new Error('can not decrease point');
+            throw new NotFoundException('user with the id is not found');
```

Comment:
In line 17, you throw `BadRequestException`, but in line 54, you throw `NotFoundException`.

Why did you separated "finding undefined user" to `BadRequestException` and `NotFoundException`?

Reply:
oh my god it's my mistake thanks

Reply:
thanks!! very nice